pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -1110,7 +1110,9 @@ public void setUrl(String url) {
     Properties p = org.postgresql.Driver.parseURL(url, null);
 
     for (PGProperty property : PGProperty.values()) {
-      setProperty(property, property.get(p));
+      if (!this.properties.containsKey(property.getName())) {
```

Comment:
i do not know is it so simple. Because now I think that default values might be not initialized at all.
Maybe should check that property unset (== null) but even then there is cap with properties with primitive type values.

Reply:
What this does is just make sure that if someone else has set the property it will not be overwritten in setURL.
Does this fix your problem ?

Reply:
yeah you are right of course. I'll check it out immediately

Reply:
Checked. This works like needed.